unstructured/ingest/connector/confluence.py

```diff
@@ -92,33 +94,85 @@ def filename(self):
             return None
         return (
             Path(self.read_config.download_dir)
-            / self.file_meta.space_id
-            / f"{self.file_meta.document_id}.html"
+            / self.document_meta.space_id
+            / f"{self.document_meta.document_id}.html"
         ).resolve()
 
     @property
     def _output_filename(self):
         """Create output file path based on output directory, space id and document id."""
-        output_file = f"{self.file_meta.document_id}.json"
-        return Path(self.partition_config.output_dir) / self.file_meta.space_id / output_file
+        output_file = f"{self.document_meta.document_id}.json"
+        return Path(self.partition_config.output_dir) / self.document_meta.space_id / output_file
+
+    @property
+    def record_locator(self) -> t.Optional[t.Dict[str, t.Any]]:
+        return {
+            "url": self.connector_config.url,
+            "page_id": self.document_meta.document_id,
+        }
+
+    @requires_dependencies(["atlassian"], extras="Confluence")
+    def _get_page(self):
+        from atlassian import Confluence
+        from atlassian.errors import ApiError
+
+        try:
+            confluence = Confluence(
+                self.connector_config.url,
+                username=self.connector_config.user_email,
+                password=self.connector_config.api_token,
+            )
+            result = confluence.get_page_by_id(
+                page_id=self.document_meta.document_id,
+                expand="history.lastUpdated,version,body.view",
+            )
+        except ApiError as e:
+            logger.error(e)
+            return None
+        return result
+
+    def update_source_metadata(self, **kwargs):
+        """Fetches file metadata from the current page."""
+        page = kwargs.get("page", self._get_page())
+        if page is None:
+            self.source_metadata = SourceMetadata(
+                exists=False,
+            )
+            return
+        document_history = page["history"]
+        date_created = datetime.strptime(
+            document_history["createdDate"],
+            "%Y-%m-%dT%H:%M:%S.%fZ",
+        ).isoformat()
+        if last_updated := document_history.get("lastUpdated", "").get("when", ""):
```

Comment:
This seems like it'll throw an `AttributeError: 'str' object has no attribute 'get'` when there's no `lastUpdated` key in `document_history` (since the default value for the first get is an empty string)

Reply:
```suggestion
        if last_updated := document_history.get("lastUpdated", {}).get("when", ""):
```